src/common/cohort_lru.h

```diff
@@ -144,7 +148,9 @@ namespace cohort {
 	    ++(o->lru_refcnt);
 	    o->lru_flags |= FLAG_EVICTING;
 	    lane.lock.unlock();
-	    if (o->reclaim()) {
+	    /* use can_replace to make sure this context holds o's partition
+	     * lock and call reclaim later*/
+	    if (o->can_replace(newobj_fac) && o->reclaim()) {
```

Comment:
so given that the current method now -must- call can_replace(fac) and then reclaim(void), is it possible that reclaim(void) should really be reclaim(fac)?  then reclaim(fac) could do the whole thing?  that seems maybe easier to understand than assuming that some object factory's can_replace() could bed a no-op

Reply:
i.e., could we change reclaim() to take an objectfactory (allowing it to fail when needed), and avoid adding can_replace(fac)?

Reply:
yes, combining can_replace(fac) and reclaim(void) is more easier to unsertand.  My thought to retain reclaim(void) is we can reuse it. Maybe there are two options, first one retain reclaim(void) and add reclaim(fac) , seond change reclaim(void) to reclaim(fac). What do you think?

Reply:
the question is, will the reclaim(void) ever be safe?  I guess if we're using the LRU without the mapping?

Reply:
I'm not sure is can be safe.  I'd say move can_replace() into reclaim()

Reply:
good point, there is a lock release cycle here